### PR TITLE
Bump OkHttp

### PR DESCRIPTION
**Pull request by jjestrel**
Update to OkHttp4

Auto converted OkHttp changes are in this commit:
https://github.com/cashapp/misk/commit/0e358947b49133595f4d822f62ecdd9432c18a74

Non-backwards compatible changes are handled in this commit:
https://github.com/cashapp/misk/commit/7ad5c689a11ad0f0fa4829e7747b9ff549a34b03

Test updates
https://github.com/cashapp/misk/pull/1094/commits/17f3375c4fe130a5ec301c865ffdb4e70624839f

### CODE REVIEW

File: misk-hibernate/src/main/kotlin/misk/hibernate/HibernateExceptionMappers.kt [hunk 1]
```diff
@@ -74,7 +75,7 @@ internal class ResourceExhaustedExceptionMapper @Inject internal constructor() :
 
 internal object ConflictExceptionResponder {
   private val HEADERS: Headers =
-      Headers.of(listOf("Content-Type" to MediaTypes.TEXT_PLAIN_UTF8).toMap())
+      listOf("Content-Type" to MediaTypes.TEXT_PLAIN_UTF8).toMap().toHeaders()
```
Comment (swankjesse) [inline misk-hibernate/src/main/kotlin/misk/hibernate/HibernateExceptionMappers.kt]: could follow-up to do just `headersOf("Content-Type", MediaTypes.TEXT_PLAIN_UTF8)`

File: misk/src/main/kotlin/misk/client/PeerClientFactory.kt [hunk 2]
```diff
@@ -81,13 +83,15 @@ class PeerClientFactory(
...
 
           return httpClientFactory.create(config).newBuilder()
-              .hostnameVerifier { _, session ->
-                val ou =
-                    (session.peerCertificates.firstOrNull() as? X509Certificate)?.let { peerCert ->
-                      X500Name.parse(peerCert.subjectX500Principal.name).organizationalUnit
-                    }
-                appName == ou
-              }
+              .hostnameVerifier(object : HostnameVerifier {
```
Comment (swankjesse) [inline misk/src/main/kotlin/misk/client/PeerClientFactory.kt]: this is the most annoying part of the Kotlin upgrade

File: misk/src/main/kotlin/misk/web/extractors/QueryParamFeatureBinding.kt [hunk 3]
```diff
@@ -31,7 +31,9 @@ internal class QueryParamFeatureBinding private constructor(
     private val name: String
   ) {
     fun bind(subject: Subject) {
-      val values = subject.httpCall.url.queryParameterValues(name)
+      val values = subject.httpCall.url.queryParameterValues(name).map {
+        it ?: throw IllegalArgumentException()
```
Comment (swankjesse) [inline misk/src/main/kotlin/misk/web/extractors/QueryParamFeatureBinding.kt]: this is correct but I think we later want to make a decision on what we do for `b` in `https://host/path?a=1&b&c=2`

File: misk/src/test/kotlin/misk/client/MockWebServerService.kt [hunk 4]
```diff
@@ -16,7 +16,9 @@ class MockWebServerService(val unixSocketFile: String?) : AbstractIdleService()
 
     server = MockWebServer()
     unixSocketFile?.let {
-      server!!.setServerSocketFactory(UnixDomainServerSocketFactory(file))
+      server!!.run {
```
Comment (swankjesse) [inline misk/src/test/kotlin/misk/client/MockWebServerService.kt]: you can probably fold the `run` call

Comment (swankjesse) [reply misk/src/test/kotlin/misk/client/MockWebServerService.kt]: (it’s caused by a limitation of the fancy deprecation feature)

Comment (jjestrel) [reply misk/src/test/kotlin/misk/client/MockWebServerService.kt]: fixed

File: misk-hibernate/src/main/kotlin/misk/hibernate/HibernateExceptionMappers.kt (hunk 1 above)
Comment (jjestrel) [reply misk-hibernate/src/main/kotlin/misk/hibernate/HibernateExceptionMappers.kt]: changed

**Pull request merged**